### PR TITLE
RXI-679 bridge filtering by issues

### PR DESCRIPTION
**Pull request by oleks-rip**

### CODE REVIEW

File: src/xbwd/client/ChainListener.cpp [hunk 1]
```diff
@@ -1239,7 +1239,7 @@ ChainListener::processTx(Json::Value const& v) const noexcept
             return warn_ret("missing or bad tx hash");
 
         auto const txnBridge = rpcResultParse::parseBridge(msg);
-        if (!txnBridge)  // TODO check bridge match
+        if (txnBridge != bridge_)
             return warn_ret("missing or bad bridge");
```
[the remark this answers is not in the document]
Comment (seelabs) [reply src/xbwd/client/ChainListener.cpp:1243]: The bridge not matching is going to be common when there are multiple bridges. I'd put this at "trace".

File: src/xbwd/client/ChainListener.cpp [hunk 2]
```diff
@@ -1738,13 +1740,8 @@ ChainListener::processNewLedger(unsigned ledgerIdx)
 }
 
 bool
-ChainListener::processBridgeReq(Json::Value const& msg)
+ChainListener::processBridgeReq(Json::Value const& msg) const noexcept
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.cpp:1743]: I don't think there's a benefit to putting `noexcept` on functions like this. There is some benefit to putting it on constructors, but I'd leave it off regular functions unless there is a reason for doing so (usually some generic code that takes different paths based on a function be `noexcept` or not).

File: src/xbwd/client/RpcResultParse.cpp [hunk 3]
```diff
@@ -19,7 +19,9 @@
 
 #include <ripple/protocol/SField.h>
 #include <ripple/protocol/jss.h>
+
 #include <xbwd/client/RpcResultParse.h>
+#include <xbwd/rpc/fromJSON.h>
```
Comment (seelabs) [inline src/xbwd/client/RpcResultParse.cpp:24]: We usually put includes in the order: project includes then library includes from most specific to most generic. (I.e. std/system goes last, boost before that, other third party deps before that, and project first). So the `xbwd` should go first.

File: src/xbwd/federator/Federator.cpp [hunk 4]
```diff
@@ -442,6 +452,16 @@ Federator::sendDBAttests(ChainType ct)
...
 
             convert(bridgeBlob, bridge, ripple::sfXChainBridge);
+            if (bridge != bridge_)
+            {
+                JLOGV(
+                    j_.warn(),
+                    "sendDBAttests bridge mismatch, skipping attestation",
+                    jv("db bridge", bridge.getJson(ripple::JsonOptions::none)),
+                    jv("current bridge",
+                       bridge_.getJson(ripple::JsonOptions::none)));
```
Comment (seelabs) [inline src/xbwd/federator/Federator.cpp:462]: This is expected, so I'd make this "trace" level.

File: src/xbwd/federator/Federator.cpp [hunk 5]
```diff
@@ -341,6 +341,16 @@ Federator::sendDBAttests(ChainType ct)
...
```
Comment (seelabs) [inline src/xbwd/federator/Federator.cpp:351]: This is expected, so I'd make this "trace" level.

**Review by seelabs — COMMENTED** [review]
Looks good, but left some nits.

File: src/xbwd/federator/Federator.cpp (hunk 4 above)
Comment (oleks-rip) [reply src/xbwd/federator/Federator.cpp:462]: I think it is not - this is tsx from the db, which shouldn't be here. Unless you change the bridge in the config.  You shouldn't switch witness to other bridge, no?

File: src/xbwd/federator/Federator.cpp (hunk 5 above)
Comment (oleks-rip) [reply src/xbwd/federator/Federator.cpp:351]: same as before

File: src/xbwd/client/ChainListener.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.cpp]: done 5b3b95ddc1b29f

File: src/xbwd/client/ChainListener.cpp (hunk 2 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.cpp]: done 5b3b95ddc1b29f

File: src/xbwd/client/RpcResultParse.cpp (hunk 3 above)
Comment (oleks-rip) [reply src/xbwd/client/RpcResultParse.cpp:21]: done 5b3b95ddc1b29f

**Review by seelabs — APPROVED** [review]
:+1:

**Review by pwang200 — APPROVED** [review]

**Comment by seelabs** [comment]
merged

**Pull request closed without merging**